packages/apps/plugins/plugin-chain/src/components/PromptTemplate/PromptTemplate.tsx

```diff
@@ -98,37 +104,30 @@ type PromptTemplateProps = {
 export const PromptTemplate = ({ prompt }: PromptTemplateProps) => {
   const { t } = useTranslation(CHAIN_PLUGIN);
   const { themeMode } = useThemeContext();
-  // TODO(burdon): Remove.
-  const model = useTextModel({ text: prompt.source });
+
+  const { doc, accessor } = useDocAccessor(prompt.source);
   const extensions = useMemo<Extension[]>(
-    () =>
-      model
-        ? [
-            createBasicExtensions({
-              bracketMatching: false,
-              lineWrapping: true,
-              placeholder: t('template placeholder'),
-            }),
-            createThemeExtensions({
-              themeMode,
-              slots: {
-                content: { className: '!p-3' },
-              },
-            }),
-            promptExtension,
-            model.extension!,
-          ]
-        : [],
-    [model, themeMode],
+    () => [
+      createDataExtensions({ id: prompt.id, text: accessor }),
+      createBasicExtensions({
+        bracketMatching: false,
+        lineWrapping: true,
+        placeholder: t('template placeholder'),
+      }),
+      createThemeExtensions({
+        themeMode,
+        slots: {
+          content: { className: '!p-3' },
+        },
+      }),
+      promptExtension,
+    ],
+    [themeMode, accessor],
```

Comment:
Should calls to `create…Extensions` always be memoized? To encourage memoization, should we provide a `useExtensions` hook that can call known create functions?

Reply:
Good catch. I'm considering making the useTextEditor take a callback fn.